app/upgrades/v16/upgrades.go

```diff
@@ -79,3 +88,38 @@ func CreateUpgradeHandler(
 		return mm.RunMigrations(ctx, configurator, vm)
 	}
 }
+
+// ConvertToNativeCoinExtensions converts all the registered ERC20 tokens of Cosmos native token pairs
+// back to the native representation and registers the (W)ERC20 precompiles for each token pair.
+func ConvertToNativeCoinExtensions(
+	ctx sdk.Context,
+	logger log.Logger,
+	accountKeeper authkeeper.AccountKeeper,
+	bankKeeper bankkeeper.Keeper,
+	erc20Keeper erc20keeper.Keeper,
+) error {
+	// NOTE (@fedekunze): first we must convert the all the registered tokens.
+	// If we do it the other way around, the conversion will fail since there won't
+	// be any contract code due to the selfdestruct.
+	if err := ConvertERC20Coins(ctx, logger, accountKeeper, bankKeeper, erc20Keeper); err != nil {
+		logger.Error("failed to convert native coins", "error", err.Error())
+		// TODO: return error here?
```

Comment:
we should return nil to not block the upgrade

Reply:
returned an error in order to catch any problems and avoid writing to the context, but it's not blocking the upgrade 👍